src/Core/Logging/Extensions/ILogServiceExtensions.Critical.cs

```diff
@@ -0,0 +1,166 @@
+            service?.Internal(
+                message,
+                ex,
+                methodName);
+        }
+
+        public static void Critical<TState>(
+            this ILogService service,
+            TState message,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                message,
+                methodName);
+        }
+
+        public static void Critical(
+            this ILogService service,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                ex,
+                methodName);
+        }
+
+
+
+        public static void Critical<TState>(
+            this ILogService service,
+            TState message,
+            Exception ex,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                message,
+                ex,
+                formatter,
+                methodName);
+        }
+
+        public static void Critical<TState>(
+            this ILogService service,
+            TState message,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                message,
+                formatter,
+                methodName);
+        }
+
+        public static void Critical(
+            this ILogService service,
+            Exception ex,
+            Func<object, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                ex,
+                formatter,
+                methodName);
+        }
+
+
+
+        public static void Critical<TState>(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            TState message,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                timestamp,
+                message,
+                formatter,
+                methodName);
+        }
+
+        public static void Critical(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            Exception ex,
+            Func<object, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service?.Internal(
+                timestamp,
+                ex,
+                formatter,
+                methodName);
+        }
+
+        public static void Critical<TState>(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            TState message,
+            Exception ex,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
```

Comment:
Codacy found an issue: [Refactor this method to add validation of parameter 'service' before using it.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5762132)